Firestore/Example/Tests/Integration/API/FIRValidationTests.mm

```diff
@@ -420,6 +420,55 @@ - (void)testQueryCannotBeCreatedFromDocumentsMissingSortValues {
   FSTAssertThrows([query queryEndingAtDocument:snapshot], reason);
 }
 
+- (void)testQueriesCannotBeSortedByAnUncommittedServerTimestamp {
+  __weak FIRCollectionReference *collection = [self collectionRef];
+  FIRFirestore *db = [self firestore];
+
+  [db disableNetworkWithCompletion:[self completionForExpectationWithName:@"Disable network"]];
+  [self awaitExpectations];
+
+  XCTestExpectation *offlineCallbackDone =
+      [self expectationWithDescription:@"offline callback done"];
+  XCTestExpectation *onlineCallbackDone = [self expectationWithDescription:@"online callback done"];
+
+  [collection addSnapshotListener:^(FIRQuerySnapshot *snapshot, NSError *error) {
+    XCTAssertNil(error);
+
+    // Skip the initial empty snapshot.
+    if (snapshot.empty) return;
+
+    XCTAssertEqual(snapshot.count, 1);
+    FIRQueryDocumentSnapshot *docSnap = [snapshot documents][0];
```

Comment:
nit: Given these are properties, I'd do `snapshot.documents[0]` and `snapshot.metadata.pendingWrites`

Reply:
Done.